src/App.js

```diff
@@ -1,22 +1,42 @@
 import "./App.css";
-// import { useState } from "react";
+import { useState } from "react";
 import { BrowserRouter, Route, Routes } from "react-router-dom";
+import { User } from "./contexts/User";
 import Header from "./componets/Header";
-import Home from "./componets/Home";
-import Article from "./componets/Article";
+import Navbar from "./componets/Navbar";
+import Profile from "./componets/Profile";
+import Homepage from "./componets/Homepage";
+import CodingArticles from "./componets/CodingArticles";
+import CookingArticles from "./componets/CookingArticles";
+import FootballArticles from "./componets/FootballArticles";
+import ArticlePage from "./componets/ArticlePage";
 
 function App() {
+  const [user, setUser] = useState({
+    username: "tickle122",
+    name: "Tom Tickle",
+    avatar_url:
+      "https://vignette.wikia.nocookie.net/mrmen/images/d/d6/Mr-Tickle-9a.png/revision/latest?cb=20180127221953",
+  });
+
   return (
     <BrowserRouter>
-      <div className="App">
-        <header className="App-header">
-          <Header />
-        </header>
-        <Routes>
-          <Route path="/" element={<Home />} />
-          <Route path="/articles/:article_id" element={<Article />} />
-        </Routes>
-      </div>
+      <User.Provider value={{ user, setUser }}>
+        <div className="App">
+          <header className="App-header">
+            <Header />
+            <Navbar />
+            <Profile />
+          </header>
+          <Routes>
+            <Route path="/" element={<Homepage />} />
+            <Route path="/topics/coding" element={<CodingArticles />} />
```

Comment:
I think if you used a parametric here rather than /topics/coding, /topics/football etc you could do your articles with a single component rather than having 3 that do essentially the same thing, or if you wanted to keep it as is, maybe you could add a topic="coding" property and use that in the component, either way I think would work ok

Reply:
I will try. Thank you!